### PR TITLE
New optional "quicklooks" flag in "download" command

### PR DESCRIPTION
**Pull request by ahuarte47**
This PR is a proposal of implementation for https://github.com/CS-SI/eodag/issues/264.

It works here for me, I hope it can be useful in some manner for your future implementation.
Thanks for developing this amazing software.

NOTE:
I have not implemented this code defining a `get_quicklook` command because of parser internally changes the parameter name to `get-quicklook`. I was not able to define the necessary extra setting.

### CODE REVIEW

**Comment by sbrunato** [comment]
Hello @ahuarte47 , many thanks for this PR following the discussion https://github.com/CS-SI/eodag/discussions/263.

I'll review it soon for a implementation in the next release

File: eodag/cli.py [hunk 1]
```diff
@@ -482,27 +488,70 @@ def download(ctx, **kwargs):
...
+
+            outputs_prefix = (
+                kwargs.get("outputs_prefix", None)
+                or download_plugin.config.outputs_prefix
             )
+            file_ext = os.path.splitext(os.path.basename(quicklook_link))[1]
+            quicklook_file = (
+                sanitize(product.properties["title"]) + "--quicklook" + file_ext
+            )
+            base_dir = outputs_prefix
```
Comment (sbrunato) [inline eodag/cli.py:519]:
```suggestion
```

File: eodag/cli.py [hunk 2]
```diff
@@ -482,27 +488,70 @@ def download(ctx, **kwargs):
...
-        for downloaded_file in downloaded_files:
-            if downloaded_file is None:
-                click.echo("A file may have been downloaded but we cannot locate it")
+            downloaded_file = product.get_quicklook(
+                filename=quicklook_file, base_dir=base_dir, progress_callback=None
+            )
+            if not downloaded_file:
+                click.echo(
+                    "A quicklook may have been downloaded but we cannot locate it"
+                )
```
Comment (sbrunato) [inline eodag/cli.py:527]:
```suggestion
            downloaded_file = product.get_quicklook()
            if not downloaded_file:
                click.echo(
                    "A quicklook may have been downloaded but we cannot locate it. "
                    "Increase verbosity for more details: `eodag -v download [OPTIONS]`"
                )
```

File: README.rst [hunk 3]
```diff
@@ -206,6 +206,10 @@ Start playing with the CLI:
 
      eodag download --search-results search_results.geojson
 
+- To download only the quicklooks of the result of the previous call to search::
```
Comment (sbrunato) [inline README.rst:209]:
```suggestion
- To download only the result quicklooks of the previous call to search::
```

**Review by sbrunato — CHANGES_REQUESTED** [review]

File: eodag/cli.py (hunk 1 above)
Comment (sbrunato) [reply eodag/cli.py:519]: The CLI should behave the same way the Python API does: downloading quicklooks in `{outputs_prefix}/quicklooks/{product_id}` (without extension). This way, the user will know in advance where the quicklook will be downloaded without having to look for the file extension or the file name

**Review by sbrunato — CHANGES_REQUESTED** [review]
Also please replace 
https://github.com/ahuarte47/eodag/blob/c09455a5ab5c9192b6b1725d5bd1ed5f8e4520f6/eodag/api/product/_product.py#L340
with
```py
        if self.properties.get("quicklook", None) is None:
```

**Review by sbrunato — APPROVED** [review]

**Comment by ahuarte47** [comment]
Thanks @sbrunato !!!!

**Pull request merged**